src/main/java/org/craftercms/studio/impl/v2/service/site/SitesServiceImpl.java

```diff
@@ -47,6 +49,12 @@ public PluginDescriptor getSiteBlueprintDescriptor(final String id) {
         return sitesServiceInternal.getSiteBlueprintDescriptor(id);
     }
 
+    @Override
+    public void validateBlueprintParameters(final PluginDescriptor descriptor, final Map<String, String> params)
```

Comment:
Why do we want to expose this API and without proper permissions?

Reply:
It needs to be exposed because it is called by `SiteServiceImpl`, I'm not sure what permissions could be applied since it is only used during create site and I think the site service handles permissions (none of the methods in `SitesServiceImpl` has permissions)

Reply:
SiteServiceImpl is V1 service. It will be deprecated at some point. SitesService(Impl) is a public service layer and should have permissions. On the other hand, parameters validation is something that should be performed either at the controller layer (as input validation) or internally within service. I don't see need exposing this API if it will be called only by create site API(s).

Reply:
So you think the method should be moved to `SiteServiceImpl` as `protected`? I place it in `SitesServiceImpl ` because all other operations related to blueprint descriptors are there